### PR TITLE
DateRangeBox: use buttonsCollection to render drop down button

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js [hunk 1]
```diff
@@ -180,4 +191,209 @@ QUnit.module('DateRangeBox Initialization', moduleConfig, () => {
...
+
+    QUnit.test('Drop down button should not be rendered if showDropDownButton is false', function(assert) {
+        this.reinit({
+            showDropDownButton: false
+        });
+
+        assert.strictEqual(getButtons(this.$element).length, 0, 'drop down button was not rendered');
+    });
+
+    QUnit.test('Drop button should not be rendered if buttons is [] and showDropDownButton is true', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: update in all tests, plz
```suggestion
    QUnit.test('Drop down button should not be rendered if buttons is [] and showDropDownButton is true', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js [hunk 2]
```diff
@@ -180,4 +191,209 @@ QUnit.module('DateRangeBox Initialization', moduleConfig, () => {
...
+        assert.strictEqual(getButtons($buttonsContainers.eq(1)).length, 1, 'drop down button was rendered in after section');
+    });
+
+    QUnit.test('getButton("dropDown") method should return instance of default drop down button', function(assert) {
+        this.reinit({});
+
+        assert.deepEqual(this.instance.getButton('dropDown'), getButtons(this.$element).dxButton('instance'));
+    });
+
+    QUnit.test('getButton("home") method should return instance of custom action button', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]:
```suggestion
    QUnit.test('getButton(name) method should return instance of custom action button by passed name', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js [hunk 3]
```diff
@@ -180,4 +191,209 @@ QUnit.module('DateRangeBox Initialization', moduleConfig, () => {
...
+        };
+
+        this.reinit({
+            dropDownButtonTemplate: buttonTemplate
+        });
+
+        assert.strictEqual(getButtons(this.$element).text(), 'Template', 'template was rendered');
+    });
+
+    QUnit.test('Drop button template should be changed correctly in runtime', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: shouldn't we also test runtime change of other props?

**Review by ksercs — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: I added tests for `buttons` & `showDropDownButton` options change in runtime https://github.com/DevExpress/DevExtreme/pull/24306/commits/0ece794aafd40022a99d12436a248615e63456f5

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: https://github.com/DevExpress/DevExtreme/pull/24306/commits/0ece794aafd40022a99d12436a248615e63456f5

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: https://github.com/DevExpress/DevExtreme/pull/24306/commits/0ece794aafd40022a99d12436a248615e63456f5

**Review by ksercs — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

**Pull request merged**